src/gui/qgssnapindicator.cpp

```diff
@@ -97,6 +97,10 @@ void QgsSnapIndicator::setMatch( const QgsPointLocator::Match &match )
     {
       iconType = QgsVertexMarker::ICON_RHOMBUS; // area snap
     }
+    else if ( match.hasLineEndpoint() )
+    {
+      iconType = QgsVertexMarker::ICON_BOX; // line endpoint snap
+    }
```

Comment:
I can think of cases where one might want to have both *Vertex* and *LineEndpoint* enabled and decide where to snap based on the displayed marker.
I think it would be a good choice to use a separate marker for LineEndpoint.

Reply:
Ok, done